### PR TITLE
[Website] Updates to homepage, highlighting features

### PR DESCRIPTION
**Pull request by EnMod**
[:mag: Preview Link](https://nomad-git-nqwebhomepage-cards-hashicorp.vercel.app/)

---

This PR replaces the "When to consider Nomad Enterprise?" section with two components:
- `<ComparisonCallouts />` - A new component displaying various points of comparison between Nomad and an alternative
- An instance of `<TextSplitWithLogoGrid />`

### Tech debt: Brand color issue for `<Button />`

**Currently, the CTA link to the left of the logo grid is not taking brand colors.**  It seems its `.variant-tertiary-neutral` class is overriding the `color` rule applied by `.brand-nomad`.  Renaming the variant class to `.variant-tertiary` solves the issue, mainly because `.variant-tertiary-neutral` doesn't factor in brand labels (as intended).

**I've hacked in an override for now in the interest of time,** but this should be looked at at the component level instead.

### CODE REVIEW

File: website/pages/home/index.jsx [hunk 1]
```diff
@@ -364,7 +365,100 @@ export default function Homepage() {
         ]}
       />
 
-      <NomadEnterpriseInfo />
+      {/* TODO Use proper URLs */}
```
Comment (EnMod) [inline website/pages/home/index.jsx:368]: Asked in Asana about these, will update when I hear back.

File: website/pages/home/style.css [hunk 2]
```diff
@@ -28,4 +28,13 @@
...
   }
+
+  /* HACK: Override variant-tertiary-neutral's color rules */
+  & .g-text-split .g-btn.variant-tertiary-neutral {
+    color: var(--nomad);
+
+    & svg path {
+      stroke: var(--nomad);
+    }
+  }
```
Comment (EnMod) [inline website/pages/home/style.css:39]: `<Button />` doesn't override its colors with the brand when tertiary it looks like, so I hacked this in given the due date 😬 This should def be addressed/considered at the component level, though.  I'm on the hook for cleaning this up when the time comes, leave it to me 😅

File: website/pages/home/index.jsx (hunk 1 above)
Comment (EnMod) [inline website/pages/home/index.jsx:368]: These are still WIP, will update when they're finalized.

File: website/pages/home/style.css (hunk 2 above)
Comment (EnMod) [inline website/pages/home/style.css:39]: `<Button />` doesn't override its colors with the brand when tertiary it looks like, so I hacked this in for now. This should def be addressed/considered at the component level, though.

**Review by mwickett — APPROVED** [review]
🎸

**Comment by mwickett** [comment]
Looks like the preview deploy didn't pick up the order change on the homepage?

**Comment by EnMod** [comment]
It's looking that way yeah.  I'm going to redeploy and see if that works

**Comment by EnMod** [comment]
@mwickett Interestingly, pushing another commit seems to have kicked off another build, as well as making the latest commit in both GH and Vercel match 🤔 🤔 🤔

**Pull request merged**